docs/rfcs/0001-designing-in-the-open.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Designing in the Open"
+weight: 1
+---
+
+# Designing in the Open
+
+* Date: 2021-11-02
+* Author: Robert Fratto (@rfratto)
+* PR: [grafana/agent#1055](https://github.com/grafana/agent/pull/1055)
+
+## Summary
+
+Traditionally, design proposals to Grafana Agent have been done in private by
+default, and then only made public after internal consensus was reached.
+This causes a few problems:
+
+* Community concerns are less likely to have an impact as core maintainers have
+  already agreed on a solution.
+* Initial design is gated to Grafana Labs employees.
+* Historical design docs with context and discussions become hard to find both
+  internally and even harder publicly.
+
+In the spirit of building a broader community, this document proposes inverting
+the design process to be public by default.
+
+## Goals
+
+* Outline a process to support public-first design documents
+* Lower the barrier to entry to becoming a maintainer that is not employed at
+  Grafana Labs
+* Encourage a community to contribute input to design documents
+* Encourage a community to contribute their own design documents
+
+## Non-Goals
+
+* Make everything public. If something can't be public for confidentiality
+  reasons, it can remain internal to the company.
+* Identify strict rules for when an RFC or issue is appropriate.
+
+## Terminology
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+[RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Proposal
+
+Public propsals come in two forms:
+
+* RFCs (i.e., this document) for larger changes
+* Issues proposals for smaller changes
+
+This document does not prescribe when something is a "larger change" or a
+"smaller change." We don't have enough experience yet to be able to
+strongly define categories, but these guidelines can be used for a gut check:
+
+* Larger changes require proposals that require thought and explore multiple
+  options, OR
+* Larger changes have a lasting effect on the project as a whole, OR
+* Larger changes touch multiple systems throughout the project or dramatic
+  changes to a single system.
+* Smaller changes are the inverse of the above.
+
+### RFCs
+
+RFCs MUST be valid markdown composed of at least the following:
+
+1. A title.
+2. A bulletpoint list of metadata that MUST include original authorship date
+   and the author's name. Author name SHOULD include their GitHub username as a
+   parenthethical.
+3. A summary section describing the context for what is being proposed.
+4. A list of goals and non-goals for the proposal.
+5. One or more sections dedicated to the proposal.
```

Comment:
A couple sections I've found useful in RFCs as technical proposals:
Drawbacks (what are the downsides to this proposal)
Rationale and Alternatives (why does the author think this is the best path forward, what other options were considered)
Prior Art (most designs aren't fully original, these can be helpful to orient discussion)
Unresolved Questions (where does the author see uncertainty/what questions should get answered before considering this a complete discussion)

I don't know how formal/strict these RFCs are likely to end up, but it could be useful to add these as suggested sections for authors who are interested in bringing a well-structured proposal forward.